src/pages/workspaces/workspace/Dashboard/CloudInformation.ts

```diff
@@ -0,0 +1,189 @@
+import { cond } from '@terra-ui-packages/core-utils';
+import { Fragment, ReactNode } from 'react';
+import { div, dl, h } from 'react-hyperscript-helpers';
+import { ClipboardButton } from 'src/components/ClipboardButton';
+import { Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { TooltipCell } from 'src/components/table';
+import { ReactComponent as GcpLogo } from 'src/images/gcp.svg';
+import { Ajax } from 'src/libs/ajax';
+import { bucketBrowserUrl } from 'src/libs/auth';
+import Events, { extractWorkspaceDetails } from 'src/libs/events';
+import { newTabLinkProps } from 'src/libs/utils';
+import {
+  AzureWorkspace,
+  canWrite,
+  GoogleWorkspace,
+  isAzureWorkspace,
+  isGoogleWorkspace,
+} from 'src/libs/workspace-utils';
+import { AzureStorageDetails } from 'src/pages/workspaces/workspace/Dashboard/AzureStorageDetails';
+import { BucketLocation } from 'src/pages/workspaces/workspace/Dashboard/BucketLocation';
+import { InfoRow } from 'src/pages/workspaces/workspace/Dashboard/InfoRow';
+import { InitializedWorkspaceWrapper as Workspace, StorageDetails } from 'src/pages/workspaces/workspace/useWorkspace';
+
+interface CloudInformationProps {
+  storageDetails: StorageDetails;
+  workspace: Workspace;
+  storageCost?: { isSuccess: boolean; estimate: string; lastUpdated?: string };
+  bucketSize?: { isSuccess: boolean; usage: string; lastUpdated?: string };
+}
+
+interface AzureCloudInformationProps extends CloudInformationProps {
+  workspace: AzureWorkspace & { workspaceInitialized: boolean };
+}
+
+interface GoogleCloudInformationProps extends CloudInformationProps {
+  workspace: GoogleWorkspace & { workspaceInitialized: boolean };
+}
+
+const AzureCloudInformation = (props: AzureCloudInformationProps): ReactNode => {
+  const { workspace, storageDetails } = props;
+  const azureContext = workspace.azureContext;
+  return [
```

Comment:
It's unusual for a component to return an array. Does this result in key warnings?

Suggest returning a Fragment instead of an array.
```suggestion
  return h(Fragment, [
```

Reply:
`ReactNode[]` is included in type `ReactNode`.

I originally broke this up from a giant ternary, so I was being conservative when changing things at the time.

Easy enough to make it return a Fragment though.